### DISCUSSION TITLE
Can it interact with the server api?

### DISCUSSION THREAD

**Discussion by jet10000** [flet-dev/flet#47] [General]
Very good, the documentation seems to only be able to construct ui at the moment.

**Comment by FeodorFitsner** [comment]
What would be your desired use case for that - could you please elaborate what API you'd like to access?

**Comment by jet10000** [comment]
I want to access server api using http and websocket.

**Comment by FeodorFitsner** [comment]
This is it: https://github.com/flet-dev/flet/blob/main/server/protocol.md

Page client - Flutter app
Host client - Python script

Let me know if this is what you are looking for. We might add it to docs/references.

**Reply by jet10000** [reply]
Yes, but the documentation doesn't mention how to use

**Reply by FeodorFitsner** [reply]
right, it's kind of "internal". Could you give me some specifics - what would you like to access/manage with the API?

**Reply by jet10000** [reply]
Thanks, I just wanted to dig a little deeper.

Waiting for you to complete the chat app sample.

**Reply by maciejjankowski** [reply]
what if I wanted to retrieve, say, weather info

**Reply by FeodorFitsner** [reply]
Absolutely!

**Reply by hanchenglong001** [reply]
Can you add an example about WebSocket？Thanks
